src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -0,0 +1,129 @@
+#pragma warning disable IDE0051
+
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract.Manifest;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Neo.SmartContract.Native.Oracle
+{
+    public sealed partial class OracleContract : NativeContract
+    {
+        private const int MaxUrlLength = 256;
+        private const int MaxFilterLength = 128;
+        private const int MaxCallbackLength = 32;
+        private const int MaxUserDataLength = 512;
```

Comment:
Add `MaxRequestedDataSize` ? or we may need to add FilterCost?

Reply:
What do you mean?

Reply:
If the requested data is too large, the oracle node will consume too much time.

Reply:
`MaxRepsonseSize` will be configured in oracle nodes.